akka-http-core/src/main/scala/akka/http/scaladsl/model/ws/UpgradeToWebSocket.scala

```diff
@@ -54,6 +57,31 @@ trait UpgradeToWebSocket extends jm.ws.UpgradeToWebSocket {
     subprotocol: Option[String]                   = None): HttpResponse =
     handleMessages(scaladsl.Flow.fromSinkAndSource(inSink, outSource), subprotocol)
 
+  def handleStrictMessages(
```

Comment:
I don't think this name quite matches what it actually does, just reading the name I'd expect non-strict messages are filtered, but in fact it turns non-strict messages into strict ones. It would be good if we could convey the `toStrict` change in the name here as well somehow.

No good name ideas from the top of my head though.

One idea that could be worth exploring, to also avoid having a combinatory explosion, would be adding some form of builder instead, the API could look something like:
```
handleWsMessages().forProto(subProto).toStrict().handleWith(flow)
handleWsMessages().forProto(subProto).only[Binary].handleWith(flow)
handleWsMessages().only[Text].handleWith(sink, source)
handleWsMessages().forProto(subProto).only[Binary].handleWith(actorRef) // for #2158
```

Reply:
Ups, strike the `toStrict` there, and strictify with default values, but maybe have a `maxStrictSize(bytes)`/`toStrictTimeout(4.seconds)` to allow customization per route.